### DISCUSSION TITLE
If you think your ESP32 is part working/slow or get URI file sending fails and you use a TTL/Serial Hat.....

### DISCUSSION THREAD

**Discussion by andygodber** [jomjol/AI-on-the-edge-device#2676] [Show and tell]
Ive been through several motherboards recently, including ones that Id ordered before and had worked.....
After returning those that clearly only had 2MB PSRAM, I found many that were technically correct, would accept the firmware and try to run the processing.
However, they would fail at the first Round and when I attempted to go to any of the web menu items it would be sooooo slooooow and I could see in the log URI failed to send messages.

Turns out the problem is some of these ESP devices have more (better?) logic pins connected (my simple language) and the motherboard hat/adapter/TTL connector interferes with them - probably generating more interrupts.

The solution is to remove/bend the unneeded pins out the way, so the the connectors look like those in the installation guide.

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/1061812/fca5622b-4659-4c4d-a89f-1d251b628270)

**Comment by GuiPerPT** [comment]
Same thing here. My device performance improved 500% after using 5v and ground instead of the ESP32-Cam programmer with the 3.3V. 
Interestingly it's only on one device, the second one looks fine with the hat.
